metricbeat/module/elasticsearch/ml_job/data.go

```diff
@@ -81,7 +81,6 @@ func eventsMapping(r mb.ReporterV2, info elasticsearch.Info, content []byte, isX
 		event.ModuleFields = common.MapStr{}
 		event.ModuleFields.Put("cluster.name", info.ClusterName)
 		event.ModuleFields.Put("cluster.id", info.ClusterID)
-		event.ModuleFields.Put("node.id", info.Name)
```

Comment:
The problem here is that we set the name instead of ID which breaks some `href` in the UI because we usually redirect with the ID. 7.16 didn't have this property set at all so removing it from now but we should populate the `node` property correctly.

See https://github.com/elastic/beats/blob/7.16/metricbeat/module/elasticsearch/ml_job/data_xpack.go#L68

Reply:
I don't see any reason to remove it. I actually don't remember why I added it but we can always cycle if necessary.

Reply:
Removed it because we write a name where the consumer of these documents expect an ID.  It currently breaks the stack monitoring UI because we build a link out of this ID to redirect to the node's details page, and the redirection fails since we can't find the node details from its name.
Can we push this and follow up with the appropriate fix that would populate the `node.id` and `name` ?

Reply:
Populated the node object by parsing the `ml_job` response.
The conditional is there because the property [is only available for opened jobs](https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-get-job-stats.html#stats-node)

Reply:
Sorry, I meant to *not* remove it. I forgot the "not" :sweat_smile: